azurelinuxagent/common/protocol/metadata.py

```diff
@@ -201,22 +200,29 @@ def get_ext_handlers(self, last_etag=None):
         }
         ext_list = ExtHandlerList()
         data, etag = self._get_data(self.ext_uri, headers=headers)
-        if last_etag == None or last_etag < etag:
+        if last_etag is None or last_etag < etag:
             set_properties("extensionHandlers", ext_list.extHandlers, data)
         return ext_list, etag
 
     def get_ext_handler_pkgs(self, ext_handler):
-        ext_handler_pkgs = ExtHandlerPackageList()
-        data = None
+        logger.info("Get extension handler packages")
+        pkg_list = ExtHandlerPackageList()
+
+        manifest = None
         for version_uri in ext_handler.versionUris:
             try:
-                data, etag = self._get_data(version_uri.uri)
+                manifest, etag = self._get_data(version_uri.uri)
+                logger.info("Successfully downloaded manifest")
                 break
             except ProtocolError as e:
-                logger.warn("Failed to get version uris: {0}", e)
-                logger.info("Retry getting version uris")
-        set_properties("extensionPackages", ext_handler_pkgs, data)
-        return ext_handler_pkgs
+                logger.warn("Failed to fetch manifest: {0}", e)
+
+        if manifest is None:
+            raise ValueError("Extension manifest is empty")
+
+        set_properties("extensionPackages", pkg_list, manifest)
```

Comment:
@hglkrijger I'm confused. How does this create an instance of ExtHandlerPackage?